packages/providers/test/provider.test.ts

```diff
@@ -651,4 +651,48 @@ describe('Provider', () => {
     );
     expect(messageProof).toMatchSnapshot();
   });
+
+  it('has default timeout of 60s', () => {
+    const provider = new Provider('http://127.0.0.1:4000/graphql');
+    expect(provider.options.timeout).toBe(60000);
+  });
+
+  it('throws AbortError on timeout when calling an operation', async () => {
+    const provider = new Provider('http://127.0.0.1:4000/graphql', { timeout: 0 });
+
+    const { error } = await safeExec(() => provider.produceBlocks(10));
+
+    expect(error).toMatchObject({
+      type: 'aborted',
+      message: 'The user aborted a request.',
+    });
+  });
+
+  // Fails because the library creates its own AbortController
+  it.skip('throws AbortError on timeout when calling a subscription', async () => {
```

Comment:
I tried implementing a timeout for subscriptions as suggested [here](https://github.com/FuelLabs/fuels-ts/blob/5310e8b5226a74058b85a6f1ca3671b6b5b5c5b7/packages/providers/fuel-core-schema.graphql#L778), but the library doesn't give a straightforward way to do so. I tried and tried, but in the end didn't succeed. I'll open up an issue for it after this gets merged. 
However, the timeout works for standard operations (non-subscriptions), as is shown in the test above.